driver/normalizer/normalizer.go

```diff
@@ -0,0 +1,213 @@
+package normalizer
+
+import (
+	"gopkg.in/bblfsh/sdk.v2/uast"
+	"gopkg.in/bblfsh/sdk.v2/uast/nodes"
+	. "gopkg.in/bblfsh/sdk.v2/uast/transformer"
+)
+
+var Preprocess = Transformers([][]Transformer{
+	{Mappings(Preprocessors...)},
+}...)
+
+var Preprocessors = []Mapping{
+	ObjectToNode{
+		InternalTypeKey: "ast_type",
+		// FIXME: if the node doesn't have positions or is missing end positions
+		// (line many native Python nodes do), the annotations of these nodes wont work
+		// with these uncommented (and see below):
+		LineKey:         "lineno",
+		ColumnKey:       "col_offset",
+		EndLineKey:      "end_lineno",
+		EndColumnKey:    "end_col_offset",
+	}.Mapping(),
+}
+
+var Normalize = Transformers([][]Transformer{
+	{Mappings(Normalizers...)},
+}...)
+
+type opFuncArguments struct {
+	args Op
+}
+
+func (op opFuncArguments) Kinds() nodes.Kind {
+	return nodes.KindArray
+}
+
+func (op opFuncArguments) Check(st *State, n nodes.Node) (bool, error) {
+	v, ok := n.(nodes.Array)
+	if !ok {
+		return false, nil
+	}
+
+	return op.args.Check(st, v)
+}
+
+func (op opFuncArguments) Construct(st *State, n nodes.Node) (nodes.Node, error) {
+	// Iterate over n.args constructing Arguments. Set the arg.Init to the values
+	// in n.defaults. Do the same for kwonlyargs/kw_defaults. Finally add vararg
+	// and kwargs if they are set
+	n, err := op.args.Construct(st, n)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// FIXME: implement
+
+	return n, nil
+}
+
+// FIXME: decorators? (annotations/tags)
+// FIXME: in Python, an argument being variadic or not depends on being on
+// args.[kwonlyargs|args] or plain "args" not on any property of the argument
```

Comment:
If I understood this correctly, it can be solved by using `Cases` operator. Please check Java/JS/Go for an example.

Reply:
It was pretty complicated. Forgot to remove that comment; I did the easiest option and changed the native structure to use the standard Arg(with properties)->Default argument instead of the multiple lists to you should have to match aligning to the right...